registration-processor/init/registration-processor-registration-status-service/src/test/java/io/mosip/registration/processor/status/api/controller/RegistrationExternalStatusControllerTest.java

```diff
@@ -0,0 +1,183 @@
+				.thenReturn("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+		when(env.getProperty("mosip.registration.processor.application.version")).thenReturn("1.0");
+		List<RegistrationStatusSubRequestDto> request = new ArrayList<>();
+		RegistrationStatusSubRequestDto regitrationid1 = new RegistrationStatusSubRequestDto();
+		RegistrationStatusSubRequestDto regitrationid2 = new RegistrationStatusSubRequestDto();
+		RegistrationStatusSubRequestDto regitrationid3 = new RegistrationStatusSubRequestDto();
+		regitrationid1.setRegistrationId("1001");
+		regitrationid2.setRegistrationId("1002");
+		regitrationid3.setRegistrationId("1003");
+		request.add(regitrationid1);
+		request.add(regitrationid2);
+		request.add(regitrationid3);
+		registrationStatusRequestDTO = new RegistrationStatusRequestDTO();
+		registrationStatusRequestDTO.setRequest(request);
+		registrationStatusRequestDTO.setId("mosip.registration.status");
+		registrationStatusRequestDTO.setVersion("1.0");
+		registrationStatusRequestDTO
+				.setRequesttime(DateUtils.getUTCCurrentDateTimeString("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"));
+		regStatusToJson = gson.toJson(registrationStatusRequestDTO);
+		registrationDtoList = new ArrayList<>();
+		registrationDtoList1 = new ArrayList<>();
+		RegistrationStatusDto registrationStatusDto1 = new RegistrationStatusDto();
+		registrationStatusDto1.setRegistrationId("1001");
+
+		RegistrationStatusDto registrationStatusDto2 = new RegistrationStatusDto();
+		registrationStatusDto2.setRegistrationId("1002");
+
+		registrationDtoList.add(registrationStatusDto1);
+		registrationDtoList1.add(registrationStatusDto2);
+
+		SearchInfo searchInfo = new SearchInfo();
+		List<FilterInfo> filterinfos = new ArrayList<FilterInfo>();
+		List<SortInfo> sortInfos = new ArrayList<SortInfo>();
+		FilterInfo filterInfo = new FilterInfo();
+		filterInfo.setColumnName("name");
+		filterInfo.setValue("mosip");
+		filterinfos.add(filterInfo);
+		SortInfo sortInfo = new SortInfo();
+		sortInfo.setSortField("createDateTime");
+		sortInfo.setSortType("desc");
+		sortInfos.add(sortInfo);
+		searchInfo.setFilters(filterinfos);
+		searchInfo.setSort(sortInfos);
+
+		Mockito.doReturn(registrationDtoList).when(registrationStatusService).getByIds(ArgumentMatchers.any());
+		Mockito.doReturn(registrationDtoList1).when(syncRegistrationService).getByIds(ArgumentMatchers.any());
+
+	}
+
+	
+	/**
+	 * Search success test.
+	 *
+	 * @throws Exception
+	 *             the exception
+	 */
+	@Test
+	public void searchSuccessTest() throws Exception {
+		doNothing().when(registrationStatusRequestValidator).validate((registrationStatusRequestDTO),
+				"mosip.registration.status");
+
+
+		this.mockMvc.perform(post("/externalstatus/search").accept(MediaType.APPLICATION_JSON_VALUE)
+				.cookie(new Cookie("Authorization", regStatusToJson)).contentType(MediaType.APPLICATION_JSON_VALUE)
+				.content(regStatusToJson.getBytes()).header("timestamp", "2019-05-07T05:13:55.704Z"))
+				.andExpect(status().isOk());
+	}
```

Comment:
can you add the assert statements for response to check if for 1001 and 1002 it has the correct status and for 1003 it has the error message

Reply:
done.

Reply:
1002 is not added to asserts

Reply:
added 1002.